src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java

```diff
@@ -1071,7 +1080,7 @@ public void initialPublish(String site, String environment, String author, Strin
                 // checkout environment branch
                 logger.debug("Checkout environment branch " + environment + " for site " + site);
                 try {
-                    git.checkout().setCreateBranch(true).setForce(true).setStartPoint("master")
```

Comment:
Just a comment: this was a bug, no? It didn't honor the config.

Reply:
I don't think it was a bug. That is published repo. The process was to pull from sandbox into published master and from there to deploy content to the environment branch. With this feature, there is an additional obstacle I had to take care off. We can discuss it for further improvements.